### DISCUSSION TITLE
Yolov8 best.pt question

### DISCUSSION THREAD

**Discussion by IgorGoldberg** [ultralytics/ultralytics#7508] [Q&A]
Hi everyone. 
1)How does the algorithm decide which epoch is the best? Is it based on MAP50 only or does it also consider the validation loss?
2) Let's say I have only the best.pt file and I have no CSV table with the results of each epoch, is there still a way for me to know what were the MAP and other metrics for the best.pt?
p.s
I have seen the discussion here,
https://github.com/ultralytics/yolov5/issues/8701
but it is regarding yolov5 and to be honest I didn't understand since we were given two different answers first you say that it is just the best MAP metric:
https://github.com/ultralytics/yolov5/issues/8701#issuecomment-1194323829
but then you say it is also based on validation loss
https://github.com/ultralytics/yolov5/issues/8701#issuecomment-1607645801

**Comment by pderrenger** [comment]
@IgorGoldberg hello! Great questions regarding the selection of the `best.pt` model during training and understanding its metrics.

1) The `best.pt` model is selected based on the highest `fitness` value that is obtained by getting a weighted sum of multiple metrics:
https://github.com/ultralytics/ultralytics/blob/8154a27e9b332dd2dac7a107003462f0e20f6179/ultralytics/utils/metrics.py#L745-L748

2) If you only have the `best.pt` file and no CSV with epoch results, you can still evaluate the model to obtain the mAP and other metrics. You can run the `val` mode using the `best.pt` file to evaluate the model on the validation dataset. Here's a concise example of how to do this:

```python
from ultralytics import YOLO

# Load your trained model
model = YOLO('best.pt')

# Evaluate the model's performance on the validation set
results = model.val(data='your_dataset.yaml')
```

This will output the mAP and other relevant metrics for your `best.pt` model. For more detailed information on the metrics and how to interpret them, you can refer to the [Val documentation](https://docs.ultralytics.com/modes/val).

I hope this clarifies your queries! If you have further questions, feel free to ask. 😊🚀
